# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546400754 28800
#      Tue Jan 01 19:45:54 2019 -0800
# Branch stable
# Node ID 09a793a92c11dc8a66d0e33bca139026fdc7d23f
# Parent  fd890481c4085f68402db75f57363abac02806fb
help.cc: Fix static analyzer detected issues (bug #55347).

* help.cc (help_system::raw_help_from_docstrings_file): Correct conditional
test and look for '\n' OR '\r' as newline characters.

diff --git a/libinterp/corefcn/help.cc b/libinterp/corefcn/help.cc
--- a/libinterp/corefcn/help.cc
+++ b/libinterp/corefcn/help.cc
@@ -625,17 +625,17 @@ namespace octave
             while (file
                    && (c = file.get ()) != std::istream::traits_type::eof ()
                    && c != '\n' && c != '\r')
               ; // skip text
 
             // skip newline characters
             while (file
                    && (c = file.get ()) != std::istream::traits_type::eof ()
-                   && c == '\n' && c == '\r')
+                   && (c == '\n' || c == '\r'))
               ; // skip text
 
             file.unget ();
 
             // Position of beginning of help text.
             std::streampos beg = file.tellg ();
 
             // Skip help text.
